cmake/FindWayland.cmake

```diff
@@ -31,29 +31,41 @@ IF (NOT WIN32)
-	FIND_LIBRARY(WAYLAND_EGL_LIBRARIES    NAMES wayland-egl      HINTS ${PKG_WAYLAND_LIBRARY_DIRS})
-	FIND_LIBRARY(WAYLAND_CURSOR_LIBRARIES NAMES wayland-cursor   HINTS ${PKG_WAYLAND_LIBRARY_DIRS})
+	FIND_LIBRARY(WAYLAND_CLIENT_LIBRARIES NAMES wayland-client)
+	FIND_LIBRARY(WAYLAND_SERVER_LIBRARIES NAMES wayland-server)
+	FIND_LIBRARY(WAYLAND_EGL_LIBRARIES    NAMES wayland-egl)
+	FIND_LIBRARY(WAYLAND_CURSOR_LIBRARIES NAMES wayland-cursor)
 
-	set(WAYLAND_INCLUDE_DIR ${WAYLAND_CLIENT_INCLUDE_DIR} ${WAYLAND_SERVER_INCLUDE_DIR} ${WAYLAND_EGL_INCLUDE_DIR} ${WAYLAND_CURSOR_INCLUDE_DIR})
+	if (WAYLAND_CLIENT_INCLUDE_DIR AND WAYLAND_CLIENT_LIBRARIES AND NOT TARGET wayland::client)
+		add_library(wayland::client STATIC IMPORTED)
+		set_target_properties(wayland::client PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${WAYLAND_CLIENT_INCLUDE_DIR}" INTERFACE_LINK_LIBRARIES "${WAYLAND_CLIENT_LIBRARIES}")
+	endif()
+	if (WAYLAND_SERVER_INCLUDE_DIR AND WAYLAND_SERVER_LIBRARIES AND NOT TARGET wayland::server)
+		add_library(wayland::server STATIC IMPORTED)
+		set_target_properties(wayland::server PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${WAYLAND_SERVER_INCLUDE_DIR}" INTERFACE_LINK_LIBRARIES "${WAYLAND_SERVER_LIBRARIES}")
+	endif()
+	if (WAYLAND_EGL_INCLUDE_DIR AND WAYLAND_EGL_LIBRARIES AND NOT TARGET wayland::egl)
+		add_library(wayland::egl STATIC IMPORTED)
+		set_target_properties(wayland::client PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${WAYLAND_EGL_INCLUDE_DIR}" INTERFACE_LINK_LIBRARIES "${WAYLAND_EGL_LIBRARIES}")
+	endif()
+	if (WAYLAND_CURSOR_INCLUDE_DIR AND WAYLAND_CURSOR_LIBRARIES AND NOT TARGET wayland::cursor)
+		add_library(wayland::cursor STATIC IMPORTED)
+		set_target_properties(wayland::cursor PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${WAYLAND_CURSOR_INCLUDE_DIR}" INTERFACE_LINK_LIBRARIES "${WAYLAND_CURSOR_LIBRARIES}")
+	endif()
 
-	set(WAYLAND_LIBRARIES ${WAYLAND_CLIENT_LIBRARIES} ${WAYLAND_SERVER_LIBRARIES} ${WAYLAND_EGL_LIBRARIES} ${WAYLAND_CURSOR_LIBRARIES})
+	set(WAYLAND_LIBRARIES wayland::client wayland::server wayland::egl wayland::cursor)
 
 	list(REMOVE_DUPLICATES WAYLAND_INCLUDE_DIR)
 
 	include(FindPackageHandleStandardArgs)
 
-	FIND_PACKAGE_HANDLE_STANDARD_ARGS(WAYLAND_CLIENT  DEFAULT_MSG  WAYLAND_CLIENT_LIBRARIES  WAYLAND_CLIENT_INCLUDE_DIR)
-	FIND_PACKAGE_HANDLE_STANDARD_ARGS(WAYLAND_SERVER  DEFAULT_MSG  WAYLAND_SERVER_LIBRARIES  WAYLAND_SERVER_INCLUDE_DIR)
-	FIND_PACKAGE_HANDLE_STANDARD_ARGS(WAYLAND_EGL     DEFAULT_MSG  WAYLAND_EGL_LIBRARIES     WAYLAND_EGL_INCLUDE_DIR)
-	FIND_PACKAGE_HANDLE_STANDARD_ARGS(WAYLAND_CURSOR  DEFAULT_MSG  WAYLAND_CURSOR_LIBRARIES  WAYLAND_CURSOR_INCLUDE_DIR)
-	FIND_PACKAGE_HANDLE_STANDARD_ARGS(WAYLAND         DEFAULT_MSG  WAYLAND_LIBRARIES         WAYLAND_INCLUDE_DIR)
+	# FIND_PACKAGE_HANDLE_STANDARD_ARGS is just meant to find the main package and set package found. Not set variables or find individual libs
+	FIND_PACKAGE_HANDLE_STANDARD_ARGS(Wayland REQUIRED_VARS WAYLAND_LIBRARIES)
```

Comment:
This won't properly fail if it fails to find wayland.  It'll look, see that `WAYLAND_LIBRARIES` is indeed set (since it's set to `wayland::client;wayland::server;wayland::egl;wayland::cursor`), and be like "yep, all good"

You want to be checking things like `WAYLAND_SERVER_INCLUDE_DIR`, `WAYLAND_SERVER_LIBRARIES`, etc